Singer.API/Services/UserProfileService.cs

```
@@ -0,0 +1,78 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Options;
+using Singer.Data;
+using Singer.Data.Models.Configuration;
+using Singer.DTOs.Users;
+using Singer.Helpers.Exceptions;
+using Singer.Models.Users;
+using Singer.Services.Interfaces;
+
+namespace Singer.Controllers
+{
+   public class UserProfileService : IUserProfileService
+   {
+      private readonly ApplicationDbContext _context;
+      private readonly UserManager<User> _userManager;
+      private readonly IEmailService<UserDTO> _emailService;
+      private readonly string _frontendUrl;
+
+      public UserProfileService(ApplicationDbContext context,UserManager<User> userManager,
+         IEmailService<UserDTO> emailService, IOptions<ApplicationConfig> applicationConfigurationOptions)
+      {
+         _context = context;
+         _userManager = userManager;
+         _emailService = emailService;
+         if (applicationConfigurationOptions == null)
+         {
+            throw new ArgumentNullException(nameof(applicationConfigurationOptions));
+         }
+         var applicationConfiguration = applicationConfigurationOptions.Value;
+         _frontendUrl = applicationConfiguration.FrontendURL;
+      }
+
+      public async Task UpdatePassword(Guid userId, string rawToken, string password)
+      {
+         var token = rawToken.Replace(" ", "+");
+         var user = await _userManager.FindByIdAsync(userId.ToString());
+         if (user == null)
+         {
+            throw new Exception("No user found");
```

Shouldn't we return an error in Dutch?